tests/python/frontend/tensorflow/test_forward.py

```diff
@@ -205,6 +212,93 @@ def test_forward_pooling():
                          dilation_rate=[1, 1],
                          strides=[2, 1])
 
+            # Tests involving SpaceToBatchND
+            _test_pooling(input_shape=[1, 1, 2, 1],
+                         window_shape=[1, 1],
+                         padding='VALID',
+                         pooling_type=pool_type,
+                         dilation_rate=[1, 2])
+
+            _test_pooling(input_shape=[1, 2, 1],
+                         window_shape=[1],
+                         padding='VALID',
+                         pooling_type=pool_type,
+                         dilation_rate=[2])
+
+
+
+#######################################################################
+# SpaceToBatchND
+# --------------
+def _test_space_to_batch_nd(data, block_shape, paddings, desired):
+    data = np.array(data)
+    input_shape = data.shape
+    input_node = relay.var('data', shape=input_shape, dtype='int32')
+    output_node = relay.frontend.tensorflow._space_to_batch_nd_impl(
+        input_node, input_shape, block_shape, paddings, params={})
+    actual = run_relay_expr(output_node, {input_node: data})
+    tvm.testing.assert_allclose(actual, desired)
+
+def test_space_to_batch_nd():
+    # test cases: https://www.tensorflow.org/api_docs/cc/class/tensorflow/ops/space-to-batch-n-d
+    _test_space_to_batch_nd(
+        data=[[[[1], [2]], [[3], [4]]]],
+        block_shape=[2, 2],
+        paddings=[[0, 0], [0, 0]],
+        desired=[[[[1]]], [[[2]]], [[[3]]], [[[4]]]]
+    )
+
+    _test_space_to_batch_nd(
+        data=[[[[1, 2, 3], [4, 5, 6]],
+               [[7, 8, 9], [10, 11, 12]]]],
+        block_shape=[2, 2],
+        paddings=[[0, 0], [0, 0]],
+        desired=[[[[1, 2, 3]]], [[[4, 5, 6]]],
+                 [[[7, 8, 9]]], [[[10, 11, 12]]]]
+    )
+
+    _test_space_to_batch_nd(
+        data=[[[[1], [2], [3], [4]],
+               [[5], [6], [7], [8]],
+               [[9], [10], [11], [12]],
+               [[13], [14], [15], [16]]]],
+        block_shape=[2, 2],
+        paddings=[[0, 0], [0, 0]],
+        desired=[[[[1], [3]], [[9], [11]]],
```

Comment:
Can you compute the desired from TF API instead ?

Reply:
Yes, I've made this change already, just not pushed yet.